File: algorithms/codewars/python/sum_two_smallest_numbers.py
"""
Create a function that returns the sum of the two lowest positive numbers given an array of minimum 4 positive integers. 
No floats or non-positive integers will be passed.

For example, when an array is passed like [19, 5, 42, 2, 77], the output should be 7.
"""
import unittest

class Solution(unittest.TestCase):
    def __init__(self):
        super().__init__()
        
    def fn(self, numbers):
        return sum(sorted(numbers)[:2])
    def test1(self):
        self.assertEqual(self.fn([5, 8, 12, 19, 22]), 13)
    def test2(self):
        self.assertEqual(self.fn([15, 28, 4, 2, 43]), 6)
    def test3(self):
        self.assertEqual(self.fn([3, 87, 45, 12, 7]), 10)
    def test4(self):
        self.assertEqual(self.fn([23, 71, 33, 82, 1]), 24)

def main():
    sol = Solution()
    sol.test1()
    sol.test2()
    sol.test3()
    sol.test4()

if __name__ == '__main__':
    main()
    print("All tests passed")

File: algorithms/codewars/sum-two-smallest-numbers/sum_two_smallest_numbers.c
/*
Create a function that returns the sum of the two lowest positive numbers given an array of minimum 4 positive integers. 
No floats or non-positive integers will be passed.

For example, when an array is passed like [19, 5, 42, 2, 77], the output should be 7.
*/
#include <stddef.h>
#include <stdio.h>
#include <stdlib.h>

long sum_two_smallest_number(size_t, const int[]);

int main(int argc, char **argv) {
    if (argc < 5) {
        printf("Incorrect number of arguments\n");
        return 1;
    }
    int numbers[argc-1];
    for (int i = 1; i < argc; i++) {
        numbers[i - 1] = strtol(argv[i], NULL, 10);
    }
    size_t n = sizeof(numbers) / sizeof(numbers[0]);
    long result = sum_two_smallest_number(argc - 1, numbers);
    printf("Result: %ld\n", result);

    return 0;
}

long sum_two_smallest_number(size_t n, const int numbers[n]) {
    int min1 = numbers[0], min2 = numbers[1];
    for (int i=2; i < n; i++) {
        if (numbers[i] < min1) {
            min2 = min1 < min2 ? min1 : min2;
            min1 = numbers[i];
        } else if (numbers[i] < min2) {
            min2 = numbers[i];
        }
    }
    
    return (long)min1 + (long)min2;
}